core/src/executor/evaluate/expr.rs

```diff
@@ -39,6 +39,10 @@ pub fn binary_op<'a>(
         }};
     }
 
+    if l.is_null() || r.is_null() {
+        return Ok(Evaluated::Value(Value::Null));
+    }
```

Comment:
Ok I see that this is necessary until we modify both `evaluate_eq` and `evaluate_cmp` return types.